plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Core.hs

```diff
@@ -1,7 +1,27 @@
-module UntypedPlutusCore.Core
-    ( module Export
-    ) where
+{-# LANGUAGE LambdaCase #-}
+
+module UntypedPlutusCore.Core (
+  module Export,
+  splitLamAbs,
+  splitApplication,
+) where
 
 import UntypedPlutusCore.Core.Instance as Export
 import UntypedPlutusCore.Core.Plated as Export
 import UntypedPlutusCore.Core.Type as Export
+
+import Data.Bifunctor
+
+-- | Strips off lambda binders.
+splitLamAbs :: Term name uni fun a -> ([name], Term name uni fun a)
+splitLamAbs = \case
+  LamAbs _ n t -> first (n :) (splitLamAbs t)
+  t            -> ([], t)
+
+-- | Strip off arguments
+splitApplication :: Term name uni fun a -> (Term name uni fun a, [(a, Term name uni fun a)])
```

Comment:
The PIR version of [splitApplication](https://github.com/input-output-hk/plutus/blob/5b7848040e932f3344ec31ff131f52371470da72/plutus-core/plutus-ir/src/PlutusIR/Contexts.hs#L27) is in a different module. In general can we be consistent with the (1) naming (2) module it belongs in for PIR and UPLC - we do want to be able to easily sync the two implementation if we need to fix a bug/add some features it would be nice to be able to easily find the counterparts.

Reply:
Maybe at least add a comment about where to find the corresponding PIR part for each of the functions.